File: SERVIDOR/PHP/modelo/Direccion.php
<?php
require_once '../controlador/conexion.php';

/**
 * 
 * 
 */
class Direccion extends Conexion{
    private $id_direccion;
    private $calle;
    private $numero;
    private $ciudad;
    private $provincia;
    private $codigo_postal;
    private $otros;
    private $coordenadas;
    
    public function getId_direccion() {
        return $this->id_direccion;
    }

    public function getCalle() {
        return $this->calle;
    }

    public function getNumero() {
        return $this->numero;
    }

    public function getCiudad() {
        return $this->ciudad;
    }

    public function getProvincia() {
        return $this->provincia;
    }

    public function getCodigo_postal() {
        return $this->codigo_postal;
    }

    public function getOtros() {
        return $this->otros;
    }

    public function getCoordenadas() {
        return $this->coordenadas;
    }

    public function setId_direccion($id_direccion): void {
        $this->id_direccion = $id_direccion;
    }

    public function setCalle($calle): void {
        $this->calle = $calle;
    }

    public function setNumero($numero): void {
        $this->numero = $numero;
    }

    public function setCiudad($ciudad): void {
        $this->ciudad = $ciudad;
    }

    public function setProvincia($provincia): void {
        $this->provincia = $provincia;
    }

    public function setCodigo_postal($codigo_postal): void {
        $this->codigo_postal = $codigo_postal;
    }

    public function setOtros($otros): void {
        $this->otros = $otros;
    }

    public function setCoordenadas($coordenadas): void {
        $this->coordenadas = $coordenadas;
    }

    public function agregar(){
        $sql = "INSERT INTO direccion (Calle, Numero, Ciudad, CP, Otros, Coordenadas) VALUES (:calle, :num, :ciu, :cp, :otros, :coor)";
        
        $sentencia = $this->dblink->prepare($sql);
        
        $calle = $this->getCalle();
        $numero = $this->getNumero();
        $ciudad = $this->getCiudad();
        $codigo_postal = $this->getCodigo_postal();
        $otros = $this->getOtros();
        $coordenadas = $this->getCoordenadas();
        
        $sentencia->bindParam(":calle", $calle);
        $sentencia->bindParam(":num", $numero);
        $sentencia->bindParam(":ciu", $ciudad);
        $sentencia->bindParam(":cp", $codigo_postal);
        $sentencia->bindParam(":otros", $otros);
        $sentencia->bindParam(":coor", $coordenadas);
        
        $resultado = $sentencia->execute();
        
        

        if ($resultado != 1) {
            //ocurrio un error al insertar
            printf('aaa');
            return FALSE;
        }

        //Insertó correctamente
        return TRUE;
    }
    
    public function agregarConId(){
        $sql = "INSERT INTO direccion (Calle, Numero, Ciudad, CP, Otros, Coordenadas) VALUES (:calle, :num, :ciu, :cp, :otros, :coor)";
        
        $sentencia = $this->dblink->prepare($sql);
        
        $calle = $this->getCalle();
        $numero = $this->getNumero();
        $ciudad = $this->getCiudad();
        $codigo_postal = $this->getCodigo_postal();
        $otros = $this->getOtros();
        $coordenadas = $this->getCoordenadas();
        
        $sentencia->bindParam(":calle", $calle);
        $sentencia->bindParam(":num", $numero);
        $sentencia->bindParam(":ciu", $ciudad);
        $sentencia->bindParam(":cp", $codigo_postal);
        $sentencia->bindParam(":otros", $otros);
        $sentencia->bindParam(":coor", $coordenadas);
        
        $resultado = $sentencia->execute();
        
        if ($resultado != 1) {
            $sql = "SELECT LAST_INSERT_ID(id) FROM direccion ORDER BY id DESC LIMIT 1";
            $sentencia = $this->dblink->prepare($sql);
            $resultado = $sentencia->execute();
            return $sentencia->fetchAll(PDO::FETCH_OBJ);
        }
        return -1;
    }
    
    public function obtenerDireccionPorId(){
        $sql = "SELECT * FROM direccion WHERE id = :id_dir";
        
        $sentencia = $this->dblink->prepare($sql);

        $id_direccion = $this->getId_direccion();
        $sentencia->bindParam(":id_dir", $id_direccion);

        $sentencia->execute();
        return $sentencia->fetchAll(PDO::FETCH_OBJ);
    }
    
}

File: CLIENTES/ESCRITORIO/proyectoPrueba/src/main/java/modelo/Direccion.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package modelo;

import com.google.gson.Gson;

/**
 *
 * @author jmcbg
 */
public class Direccion {

    private int id_direccion;
    private String calle;
    private int numero;
    private String ciudad;
    private String otros;
    private int codigo_postal;
    private int coordenadas;
    private int id_imagen;
    private boolean valida;

    public boolean isValida() {
        return valida;
    }

    public void setValida(boolean valida) {
        this.valida = valida;
    }

    public int getId_imagen() {
        return id_imagen;
    }

    public void setId_imagen(int id_imagen) {
        this.id_imagen = id_imagen;
    }

    public Direccion(int id_direccion, String calle, int numero, String ciudad, String otros, int codigo_postal, int coordenadas) {
        this.id_direccion = id_direccion;
        this.calle = calle;
        this.numero = numero;
        this.ciudad = ciudad;
        this.otros = otros;
        this.codigo_postal = codigo_postal;
        this.coordenadas = coordenadas;
    }

    public Direccion(String calle, int numero, String ciudad, String otros, int codigo_postal, int coordenadas) {
        this.calle = calle;
        this.numero = numero;
        this.ciudad = ciudad;
        this.otros = otros;
        this.codigo_postal = codigo_postal;
        this.coordenadas = coordenadas;
    }

    public Direccion(int id_direccion) {
        this.id_direccion = id_direccion;
    }

    public Direccion() {

    }

    public int getId_direccion() {
        return id_direccion;
    }

    public void setId_direccion(int id_direccion) {
        this.id_direccion = id_direccion;
    }

    public String getCalle() {
        return calle;
    }

    public void setCalle(String calle) {
        this.calle = calle;
    }

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public String getCiudad() {
        return ciudad;
    }

    public void setCiudad(String ciudad) {
        this.ciudad = ciudad;
    }

    public String getOtros() {
        return otros;
    }

    public void setOtros(String otros) {
        this.otros = otros;
    }

    public int getCodigo_postal() {
        return codigo_postal;
    }

    public void setCodigo_postal(int codigo_postal) {
        this.codigo_postal = codigo_postal;
    }

    public int getCoordenadas() {
        return coordenadas;
    }

    public void setCoordenadas(int coordenadas) {
        this.coordenadas = coordenadas;
    }

    public String getJSON() {
        return new Gson().toJson(this);
    }
}
